networkx/classes/graph.py

```diff
@@ -905,7 +890,10 @@ def add_weighted_edges_from(self, ebunch, weight='weight', **attr):
         >>> G = nx.Graph()   # or DiGraph, MultiGraph, MultiDiGraph, etc
         >>> G.add_weighted_edges_from([(0,1,3.0),(1,2,7.5)])
         """
-        self.add_edges_from(((u, v, {weight: d}) for u, v, d in ebunch), **attr)
+        # TODO For some reason this fails when `ebunch` is an iterator.
```

Comment:
It passes all tests on my local machine. @jfinkels What was the error you were facing?